pkg/daemon/update.go

```diff
@@ -1764,6 +1764,27 @@ func fileExists(path string) (bool, error) {
 	return false, fmt.Errorf("cannot stat file: %w", err)
 }
 
+// Determines if a directory exists by checking the returned error when we stat the file.
+// Also, check that it is a directory.
+func directoryExists(path string) (bool, error) {
```

Comment:
I'm not objecting to this, but I don't personally think it's worth carrying over just using `os.MkdirAll()` from the standard library like we do in other places (a *lot* of other places really).

Reply:
good point, doing all these additional check isn't adding much value. Replaced with using MkdirAll()